javaslang/src/main/java/javaslang/collection/Array.java

```diff
@@ -546,24 +558,22 @@ public T get(int index) {
     public Array<T> drop(long n) {
         if (n <= 0) {
             return this;
-        }
-        if (n >= length()) {
+        } else if (n >= length()) {
             return empty();
+        } else {
+            return with(startIndex + (int) n, length - (int) n);
```

Comment:
Ok, I think it is the best we can do. Segmenting using a two-dimensional array would be too much...